<launch>
 
  <!-- these are the arguments you can pass this launch file, for example paused:=true -->
  <arg name="paused" default="true"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="gui" default="true"/>
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/>
 
  <!-- We resume the logic in empty_world.launch -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <!-- <arg name="world_name" default="$(find gjt_ur_moveit_gazebo)/worlds/demo.world"/> -->
    <arg name="debug" value="$(arg debug)" />
    <arg name="gui" value="$(arg gui)" />
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="headless" value="$(arg headless)"/>
  </include>
  
  <!-- Load the URDF into the ROS Parameter Server -->
  <!-- <include file="$(find gjt_ur_description)/launch/ur5_upload_robotiq85_gripper.launch"/> -->
  <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find gjt_ur_description)/urdf/ur5_robot_robotiq85_gripper.urdf.xacro'" /> 
    <node name="spawn_gazebo_model" pkg="gazebo_ros" type="spawn_model" args="-urdf -param robot_description -model ur5_gripper 
    -J elbow_joint 1.3118
    -J shoulder_lift_joint -1.7605
    -J shoulder_pan_joint   0
    -J wrist_1_joint -1.122
    -J wrist_2_joint  -1.57
    -J wrist_3_joint  -1.54" respawn="false" output="screen" />
     <!-- [elbow_joint, gripper_finger1_joint, shoulder_lift_joint, shoulder_pan_joint, wrist_1_joint,wrist_2_joint, wrist_3_joint] -->


  <!-- <node name="rqt_image_view_node" pkg="rqt_image_view" type="rqt_image_view" /> -->

<!-- rviz :src/gjt_ur5_gripper_moveit_config/launch/moveit_planning_execution.launch -->
<!-- gz sdf -p /model.urdf > /model.sdf -->
<!--models-->
  <!-- <node name="base_table" pkg="gazebo_ros" type="spawn_model" args="-file $(find gjt_ur_gazebo)/urdf/cafe_table/model.sdf -sdf -model base_table -x 0.01 -y 0.65 -z 0.0" /> -->
  <node name="base_table" pkg="gazebo_ros" type="spawn_model" args="-file $(find gjt_ur_gazebo)/urdf/short_table/model.sdf -sdf -model base_table -x -0.1  -y 0 -z 0.0" /> 
 <!-- <node name="coke" pkg="gazebo_ros" type="spawn_model" args="-file $(find gjt_ur_gazebo)/urdf/coke_can/model.sdf -sdf -model coke -x 0 -y 0.7 -z 1.1 -R 1.57 -Y -0.3665" /> -->
  <!-- <node name="dropbox" pkg="gazebo_ros" type="spawn_model" args="-file $(find gjt_ur_gazebo)/urdf/dropbox.urdf -urdf -model dropbox -x 0.43 -y 0.1 -z 0" /> -->
  <!-- <node name="rectangle" pkg="gazebo_ros" type="spawn_model" args="-file $(find gjt_ur_gazebo)/urdf/rectangle.sdf -sdf -model rectangle -x 0.0 -y 0.70 -z 0.52" />  -->
  <!-- <node name="banana" pkg="gazebo_ros" type="spawn_model" args="-file $(find gjt_ur_gazebo)/urdf/banana.urdf -urdf -model banana -x 0.02 -y 0.57 -z 0.8 -R 1.57 -Y 2.2" /> -->
  <!-- <node name="green_cylinder" pkg="gazebo_ros" type="spawn_model" args="-file $(find gjt_ur_gazebo)/urdf/cylinder.urdf -urdf -model green_cylinder -x 0 -y 0.7 -z 1.2" /> -->
  <!-- <node name="red_box" pkg="gazebo_ros" type="spawn_model" args="-file $(find gjt_ur_gazebo)/urdf/block.urdf -urdf -model red_box -x -0.2 -y 0.55 -z 1.2 " /> 
  <node name="pink_box" pkg="gazebo_ros" type="spawn_model" args="-file $(find gjt_ur_gazebo)/urdf/block1.urdf -urdf -model pink_box -x 0.3 -y 0.4 -z 1 " />  -->
  
</launch>

